pages/portal/resources/[id].tsx

```diff
@@ -0,0 +1,182 @@
+import { NextPage, NextPageContext } from 'next'
+import Head from "next/head";
+import React, { useRef, FormEvent } from 'react'
+import { Resource } from 'utils/types'
+import { getResources } from "requests/Resource";
+import Navigation from "components/Portal/Navigation";
+import Router from 'next/router';
+import urls from 'utils/urls'
+
+interface Props {
+    resource: Resource;
+}
+
+
+const editResourcePage: NextPage<Props> = ({resource}) => {
+
+    const nameEle = useRef<HTMLInputElement>(null)
+    const cateEle = useRef<HTMLSelectElement>(null)
+    const linkEle = useRef<HTMLInputElement>(null)
+
+    const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+
+        const chagedResource: Resource = {
+            _id: resource._id,
+            name: nameEle.current?.value,
+            category: cateEle.current?.value,
+            link: linkEle.current?.value,
+        }
+
+        const response = await fetch(`${urls.baseUrl}${urls.api.resource.update}`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(chagedResource)
+        })
+
+        Router.push('/portal/resources')
+    }
+
+
+    return (
+        <div className="container">
+
+        <Head>
+            <title>MindVersity | Admin Portal</title>
+            <link rel="icon" href="/favicon.ico" />
+        </Head>
+
+        <Navigation />
+
+        <div className="bodyContent">
+                <h1>Edit Resource</h1>
+                <div className="formContainer">
+                    <form onSubmit={handleSubmit} method="post">
+                        <label htmlFor="name">Resource Name</label>
+                        <input ref={nameEle} type="text" name="name" placeholder="Resource Name" defaultValue={resource.name} required/>
+                        <label htmlFor="category">Category</label>
+                        <select ref={cateEle} name="category" id="categoryChoice" defaultValue={resource.category || "National"}>
```

Comment:
defaultValue is case sensitive, so "National" is not actually pointing to the option with value="national". This does not really matter since the national option is first and thus is naturally the default, but I thought I'd just point this out.

Reply:
Good catch! It should be fixed now